test: move accounting_tests and rpc_wallet_tests to wallet/test

laanwj opened the pull request:
Move the two other wallet tests to where they belong.

@jonasschnelli I'm surprised you didn't stumble on this one yet

laanwj committed:
test: move accounting_tests and rpc_wallet_tests to wallet/test

Move the two other wallet tests to where they belong.

jonasschnelli commented:
Yes! This should have been done long ago. Thanks for fixing.
utACK de39c95c2468cc643a5d5bed9e5dd8cea6a40747

laanwj commented:
With this, I guess we also want a wallet-specific test fixture, to get rid of the `ENABLE_WALLET`s in `test_bitcoin.cpp`?

laanwj committed:
test: Create test fixture for wallet

Removes all the `#ifdef ENABLE_WALLET` from `test_bitcoin` by
making the wallet tests use their own fixture.

laanwj commented:
Done. Latest commit removes all wallet-related setup from `test_bitcoin.cpp`. Wallet tests set up and tear down their own environment.

These references to the wallet remain in `src/test`, this probably signifies some code movement is still necessary:
```
multisig_tests.cpp:#include "wallet/wallet_ismine.h"
script_P2SH_tests.cpp:#include "wallet/wallet_ismine.h"
```

sipa commented:
I believe the wallet_ismine code should be moved to script/sign or
script/standard.

MarcoFalke commented on src/wallet/test/wallet_test_fixture.cpp:
```diff
@@ -0,0 +1,28 @@
+    bitdb.MakeMock();
+
+    bool fFirstRun;
+    pwalletMain = new CWallet("wallet.dat");
```
Could make sense to use "wallet_test.dat" here, to indicate that the file name is not hardcoded?

laanwj replied:
Sounds good to me in principle (although I just moved this code here, I didn't touch it)

laanwj commented:
@sipa What about keeping it separate and just moving, e.g. `script/ismine.cpp` and `script/ismine.h`?

laanwj committed:
wallet_ismine.h → script/ismine.h

Removes conditional dependency of `src/test` on wallet.

Makes multisig and P2SH tests complete without wallet built-in.

laanwj committed:
test: Rename wallet.dat to wallet_test.dat

Indicate that the file name is not hardcoded, and a little bit of safety
so that it never nukes the main wallet.

Suggestion by Marco Falke.

laanwj commented:
> moving, e.g. script/ismine.cpp and script/ismine.h?

Done. All conditional wallet stuff is gone now from `src/test`.

MarcoFalke commented:
Confirmed move only besides b30fb42

utACK b30fb42
